File: python/19-tangled-web/src/config_loader.py
import json, os, datetime

class ConfigLoader:
    def __init__(self):
        pass

    def load_config(self):
        defaults = {"retries": 3, "theme": "standard", "discount": 0}
        try:
            path = os.environ.get("APP_CONFIG_PATH", "/tmp/config.json")
            with open(path) as f:
                local = json.load(f)
        except Exception:
            local = {}

        env_config = {}
        if os.environ.get("APP_THEME"):
            env_config["theme"] = os.environ["APP_THEME"]
        if os.environ.get("APP_RETRIES"):
            env_config["retries"] = int(os.environ["APP_RETRIES"])

        seasonal = {}
        month = int(os.environ.get("APP_CURRENT_MONTH", datetime.datetime.now().month))
        if month == 12 or month == 1 or month == 2:
            seasonal = {"theme":"winter","discount":0.1}
        elif month >= 6 and month <= 8:
            seasonal = {"theme":"summer","discount":0.05}
        result = {}
        result.update(defaults); result.update(local); result.update(env_config); result.update(seasonal)
        return result

File: csharp/19-tangled-web/src/config_loader.cs
using System.Text.Json;

public class ConfigLoader
{
    public Dictionary<string, object> load_config()
    {
        Dictionary<string, object> defaults = new()
        {
            ["retries"] = 3,
            ["theme"] = "standard",
            ["discount"] = 0.0
        };
        Dictionary<string, object> local = new();
        try
        {
            var path = Environment.GetEnvironmentVariable("APP_CONFIG_PATH") ?? "/tmp/config.json";
            var parsed = JsonDocument.Parse(File.ReadAllText(path)).RootElement;
            foreach (var property in parsed.EnumerateObject())
            {
                local[property.Name] = property.Value.ValueKind == JsonValueKind.Number
                    ? property.Value.GetDouble()
                    : property.Value.ToString();
            }
        }
        catch
        {
        }

        Dictionary<string, object> envConfig = new();
        if (Environment.GetEnvironmentVariable("APP_THEME") is string theme)
        {
            envConfig["theme"] = theme;
        }

        if (Environment.GetEnvironmentVariable("APP_RETRIES") is string retries)
        {
            envConfig["retries"] = int.Parse(retries);
        }

        var month = int.Parse(Environment.GetEnvironmentVariable("APP_CURRENT_MONTH") ?? DateTime.Now.Month.ToString());
        Dictionary<string, object> seasonal = new();
        if (month == 12 || month == 1 || month == 2)
        {
            seasonal["theme"] = "winter";
            seasonal["discount"] = 0.1;
        }
        else if (month >= 6 && month <= 8)
        {
            seasonal["theme"] = "summer";
            seasonal["discount"] = 0.05;
        }

        var result = new Dictionary<string, object>();
        foreach (var source in new[] { defaults, local, envConfig, seasonal })
        {
            foreach (var pair in source)
            {
                result[pair.Key] = pair.Value;
            }
        }

        return result;
    }
}
